sdk/src/types/block/output/mod.rs

```diff
@@ -131,6 +131,18 @@ impl OutputWithMetadata {
     }
 }
 
+/// Contains the generic [`Output`] and the associated [`OutputMetadata`].
+#[derive(Clone, Debug, Eq, PartialEq)]
+#[cfg_attr(
+    feature = "serde",
+    derive(serde::Serialize, serde::Deserialize),
+    serde(rename_all = "camelCase")
+)]
+pub struct OutputAndMetadata {
```

Comment:
With/And ? Why both?

Reply:
I think we can get rid of that type, but we probably should discuss it. `OutputWithMetadata` contains the `OutputIdProof` (see TIP-48) in the response, but currently there are still places where the proof isn't needed. I could have used a tuple, and maybe I will, or we use `OutputWithMetadata` in those places as well. Will get clear on that today...

Reply:
The problem is basically that `TransactionWithMetadata` doesn't contain the proofs for its `inputs`. And we're coming from `InputSigningData` which we currently map to something that contains the `output` and its `output_id`, but NOT its proof. See here: https://github.com/iotaledger/iota-sdk/blob/2.0/sdk/src/wallet/operations/transaction/mod.rs#L171

Reply:
not part of this PR anymore